### PR TITLE
New discussions index view

### PR DESCRIPTION
**Pull request by felipecalvo**
## :dart: Goal

Redesign the discussions index on forum.

## :memo: Details

* Information about who initiated the discussion (and when) are added to each card.

* We now show both total number of messages _and_ total number of verified messages, instead of an ambiguous icon which showed verified messages count only.

* Moderator avatar + initials are replaced by a `taking care of` badge.

* Better support for custom CSS, so we can easily avoid changing the color of everything in the index when we change link colors

* When creating a discussion, the submit button now says `Publicar consulta` instead of `Guardar`.

### Before

image

_______

### After - General view

image

_______

### After - Mobile view

image
______

### :spiral_notepad: Note to reviewers

Because I designed this on-the-fly, a couple of commits do stuff that was later reversed. I'd normally try to rebase and remove those but it didn't prove too easy.

I also moved the whole discussions index to a new file, so history is a little hard to follow. I'd suggest looking at the final result instead of reviewing on a commit-per-commit basis.

### CODE REVIEW

File: app/assets/stylesheets/mumuki_laboratory/application/modules/_discussion.scss [hunk 1]
```diff
@@ -106,11 +98,30 @@ $moderator-badge-color: #dd9900;
...
   .discussion-description {
     overflow: hidden;
     display: block;
     text-overflow: ellipsis;
     white-space: nowrap;
+    margin-bottom: 20px;
+    font-size: calc(#{$font-size-base} * 0.95);
+  }
+  .discussion-initiator {
+    font-size: calc(#{$font-size-base} * 0.85);
```
Comment (felipecalvo) [inline app/assets/stylesheets/mumuki_laboratory/application/modules/_discussion.scss:120]: I *really* want to shrink our base font size by 1px. It'll still be clear and easy to read while giving us a little bit more space to work with, especially on mobile screens. Using this variable lets me not-break-everything if I finally change the font size.

File: app/helpers/discussions_helper.rb [hunk 2]
```diff
@@ -60,11 +60,18 @@ def forum_terms_link
...
+      <span class="discussion-messages-count">
+        #{fa_icon :comments, type: :regular, text: discussion.messages_count}
+      </span>
+    }.html_safe
+  end
+
+  def discussion_validated_messages_count(discussion)
+    %Q{
+      <span class="discussion-validated-messages-count">
+        #{fa_icon :comment, type: :regular}#{fa_icon :check, text: discussion.validated_messages_count}
```
Comment (felipecalvo) [inline app/helpers/discussions_helper.rb:74]: Wrote this on the commit message, but a newline here would add an undesired whitespace on the HTML. There might be a cleaner way, but just sticking both icons together seemed to work just fine

File: app/views/layouts/_discussions.html.erb [hunk 3]
```diff
@@ -40,43 +40,7 @@
...
-                <span class="discussion-description">
-                  <%= discussion.description %>
-                </span>
-              </div>
-            </div>
-          </div>
-        <% end %>
-      <% end %>
-    </div>
+    <%= render partial: 'layouts/discussions_list' %>
```
Comment (felipecalvo) [inline app/views/layouts/_discussions.html.erb:43]: I extracted all of this to a partial because we had this code duplicated _somewhere else_ and I intend to reuse it.

File: app/views/layouts/_discussions_list.html.erb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="discussions">
   <% @filtered_discussions.each do |discussion| %>
-    <%= link_to exercise_discussion_path(discussion.exercise.id, discussion) do %>
+    <%= link_to exercise_discussion_path(discussion.exercise.id, discussion), class: 'discussion-link' do %>
```
Comment (felipecalvo) [inline app/views/layouts/_discussions_list.html.erb]: I wrote this on the commit message too, but here it goes again:

Most organizations change the link colors through custom CSS.
Since each discussion on the index is wrapped in an `<a>` tag,
all text would change to the link color, which doesn't look very
good. Now that the tag can be easily identified, link colors can
be changed using `a:not(.discussion-link)`.

File: app/assets/stylesheets/mumuki_laboratory/application/modules/_discussion.scss (hunk 1 above)
Comment (flbulgarelli) [reply app/assets/stylesheets/mumuki_laboratory/application/modules/_discussion.scss:120]: I am not asking to change this here, but I'd really recommend reinforcing the policy of adding `mu-` prefix to new css classes as opposed to legacy css code which lacked of good semantics, was difficult to customize and allowed name-clashes.

File: lib/mumuki/laboratory/locales/en.yml [hunk 5]
```diff
@@ -259,7 +260,8 @@ en:
   processing_your_solution: We are processing you solution
   profile_of: Profile of %{username}
   programming_since: Started programming
-  progress: Progresss
+  progress: Progress
```
Comment (flbulgarelli) [inline lib/mumuki/laboratory/locales/en.yml:263]: :rofl:

File: lib/mumuki/laboratory/locales/es-CL.yml [hunk 6]
```diff
@@ -11,14 +11,16 @@ es-CL:
   and: y
   appendix: Apéndice
   appendix_teaser: ¿Quieres saber más? <a href="%{link}">Consulta el apéndice de este capítulo</a>
-  are_you_sure: '¿Estás seguro que quieres %{action}?'
-  ask_a_question: ¡Haz una pregunta!
+  approved_message: Validado por mentor
```
Comment (flbulgarelli) [inline lib/mumuki/laboratory/locales/es-CL.yml:14]: Although this translation predates this PR, I don't like the fact that it is gender-biased. Perhaps we could use _Validado por mentorías_. We should check that with @Rgonzalezt or @lauramangifesta

File: lib/mumuki/laboratory/locales/es.yml [hunk 7]
```diff
@@ -15,14 +15,15 @@ es:
   appendix: Apéndice
   appendix_teaser: ¿Querés saber más? <a href="%{link}">Consultá el apéndice de este capítulo</a>
   approved_message: Validado por mentor
-  are_you_sure: '¿Estás seguro que querés %{action}?'
-  ask_a_question: ¡Hacé una pregunta!
   approved_messages:
     one: validado
     other: validados
+  are_you_sure: '¿Estás seguro que querés %{action}?'
```
Comment (flbulgarelli) [inline lib/mumuki/laboratory/locales/es.yml:21]: Same comment as before. We should paraphrase this as _¿Confirmás que querés ....?_ or  something like that.

File: lib/mumuki/laboratory/locales/es-CL.yml (hunk 6 above)
Comment (rgonzalezt) [reply lib/mumuki/laboratory/locales/es-CL.yml:14]: Yes, i like that suggestion. We could use "Validado por mentorías" instead of "Validado por mentor".

File: lib/mumuki/laboratory/locales/es.yml (hunk 7 above)
Comment (rgonzalezt) [reply lib/mumuki/laboratory/locales/es.yml:21]: Yes, that would be nice!

File: app/assets/stylesheets/mumuki_laboratory/application/modules/_discussion.scss (hunk 1 above)
Comment (felipecalvo) [reply app/assets/stylesheets/mumuki_laboratory/application/modules/_discussion.scss:120]: Since the whole file is full with classes that lack the prefix, I wrote this issue that we can tackle later. https://github.com/mumuki/mumuki-laboratory/issues/1646

**Comment by felipecalvo** [comment]
Made those locales gender-neutral as suggested.
